src/iter/test.rs

```diff
@@ -1608,3 +1608,14 @@ fn check_partition_map() {
     assert_eq!(a, vec![1, 2, 3]);
     assert_eq!(b, "abcxyz");
 }
+
+#[test]
+fn check_repeat_eq() {
+    let v = vec!(4,4,4,4);
+    let mut fours: Vec<_> = repeat(4)
+    .take(4)
+    .zip(v)
+    .collect();
```

Comment:
Since your producer's `split_at` doesn't actually use the index, I don't think this `take(4)` actually works -- but the result is still limited by the `zip` length instead.  Try another test with just `repeat(4).take(4).collect()`, and it panics "too many values pushed to consumer".